### PR TITLE
esp32s2: Fix SPI

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Fix issues found on ESP32S2 SPI driver
## Impact
User will be able to use SPI on ESP32S2
## Testing
esp32s2-saola-1

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/xtensa/src/esp32s2/esp32s2_spi.c [hunk 1]
```diff
@@ -233,11 +352,18 @@ static struct esp32s2_spi_priv_s esp32s2_spi2_priv =
 {
   .spi_dev     =
     {
-      .ops     = &esp32s2_spi2_ops
+      .ops = &esp32s2_spi2_ops
     },
   .config      = &esp32s2_spi2_config,
   .refs        = 0,
-  .lock        = NXMUTEX_INITIALIZER,
+  .exclsem     = SEM_INITIALIZER(0),
```
Comment (masayuki2009) [inline arch/xtensa/src/esp32s2/esp32s2_spi.c:359]: @acassis 
Do you want to revert the change introduced https://github.com/apache/incubator-nuttx/pull/6965?
I think the changes are not DMA related and perhaps your code base is too old.
